### PR TITLE
Streaming endpoint buffer

### PR DESCRIPTION
**Pull request by joseluis-fw**
This fixes the currency pair rate being reused over and over no matter the value read in the HTTP chunk.

We have been using it like this in production for months.

Sorry for the late PR.

### CODE REVIEW

**Comment by joseluis-fw** [comment]
@kobusjoubert any suggestion here?

File: lib/oanda_api_v20/pricing.rb [hunk 1]
```diff
@@ -8,26 +8,29 @@ def pricing(options)
...
+        cleaned_message = message.strip
+        next if cleaned_message.empty?
+        yield JSON.parse(cleaned_message)
+      end
+    rescue JSON::ParserError => e
+      raise OandaApiV20::ParseError, "#{e.message} in '#{fragment}'"
+    ensure
+      buffer.clear
+    end
+
```
Comment (kobusjoubert) [inline lib/oanda_api_v20/pricing.rb]: Can we maybe just remove this empty line here. Amazing work, thank you so much. The rest is perfect!

**Comment by kobusjoubert** [comment]
> This fixes the currency pair rate being reused over and over no matter the value read in the HTTP chunk.
> 
> We have been using it like this in production for months.
> 
> Sorry for the late PR.

I just realised this pull request has been open for almost a month now. I'm really sorry for only coming back to it now, I really didn't an email notification from GitHub :(

I only received an email notification from GitHub yesterday when you mentioned @kobusjoubert, so that helped. Thanks @joseluis-fw

**Comment by josacar** [comment]
Thank you too @kobusjoubert

**Comment by kobusjoubert** [comment]
You're welcome, thanks guys for your help on this gem! @josacar @joseluis-fw

**Pull request merged**